qa/ccs-rolling-upgrade-remote-cluster/src/test/java/org/elasticsearch/upgrades/AggregationsIT.java

```diff
@@ -0,0 +1,185 @@
+                createindex(localClient, localIndex);
+                createindex(remoteClient, remoteIndex);
+            }
+        }
+    }
+
+    @After
+    private void clearClusters() throws Exception {
+        try (RestClient localClient = newLocalClient(); RestClient remoteClient = newRemoteClient()) {
+            deleteIndex(localClient, localIndex);
+            deleteIndex(remoteClient, remoteIndex);
+            docs = 0;
+        }
+    }
+
+    private void createindex(RestClient client, String index) throws IOException {
+        final String mapping = """
+             "properties": {
+               "date": { "type": "date" },
+               "number": { "type": "integer" }
+             }
+            """;
+        createIndex(client, index, Settings.EMPTY, mapping);
+        for (int i = 0; i < docs; i++) {
+            Request createDoc = new Request("POST", "/" + index + "/_doc/id_" + i);
+            createDoc.setJsonEntity(Strings.format("""
+                { "date": %s, "number": %s }
+                """, i * 1000 * 60, i));
+            assertOK(client.performRequest(createDoc));
+        }
+        refresh(client, index);
+    }
+
+    private static void configureRemoteClusters(RestClient localClient, List<SearchStatesIT.Node> remoteNodes) throws Exception {
+        final String remoteClusterSettingPrefix = "cluster.remote." + CLUSTER_ALIAS + ".";
+        final Settings remoteConnectionSettings;
+        final List<String> seeds = remoteNodes.stream()
+            .filter(n -> n.attributes().containsKey("gateway"))
+            .map(n -> n.transportAddress())
+            .collect(Collectors.toList());
+        remoteConnectionSettings = Settings.builder()
+            .putNull(remoteClusterSettingPrefix + "proxy_address")
+            .put(remoteClusterSettingPrefix + "mode", "sniff")
+            .putList(remoteClusterSettingPrefix + "seeds", seeds)
+            .build();
+        updateClusterSettings(localClient, remoteConnectionSettings);
+        assertBusy(() -> {
+            final Response resp = localClient.performRequest(new Request("GET", "/_remote/info"));
+            assertOK(resp);
+            final ObjectPath objectPath = ObjectPath.createFromResponse(resp);
+            assertNotNull(objectPath.evaluate(CLUSTER_ALIAS));
+            assertTrue(objectPath.evaluate(CLUSTER_ALIAS + ".connected"));
+        }, 60, TimeUnit.SECONDS);
+    }
+
+    public void testDateHistogram() throws Exception {
+        for (int i = 0; i < 3; i++) {
+            try (RestClient localClient = newLocalClient()) {
+                String in = URLEncoder.encode(localIndex + ",remote_cluster:" + remoteIndex, StandardCharsets.UTF_8);
+                Request request = new Request("POST", "/" + in + "/_search");
+                request.setJsonEntity(
+                    "{\"aggs\": { \"hist\": { \"date_histogram\": { \"field\": \"date\", \"calendar_interval\": \"minute\" }}}}"
```

Comment:
Minor thing, probably easier to read if `"""`.

Reply:
I'll make it nicer when push it to main

Reply:
No big deal either way. I'm sure we have plenty of `\"`.